src/lib/common/sol-certificate-impl-linux.c

```diff
@@ -31,6 +32,7 @@ struct sol_cert {
     int refcnt;
 
     char *filename;
+    char *basename;
```

Comment:
to make it clear, add `const` to basename, so we're clear that shouldn't be touched or freed, since it's inside filename